worlds/sc2/rules.py

```diff
@@ -427,7 +576,10 @@ def terran_respond_to_colony_infestations(self, state: CollectionState) -> bool:
             and self.terran_competent_anti_air(state)
             and (
                     self.terran_air_anti_air(state)
-                    or state.has_any({item_names.BATTLECRUISER, item_names.VALKYRIE}, self.player)
+                    or (
+                        state.has_any({item_names.BATTLECRUISER, item_names.VALKYRIE}, self.player)
+                        and self.weapon_armor_upgrade_count(item_names.PROGRESSIVE_TERRAN_SHIP_WEAPON, state) >= 2
```

Comment:
Again, not sure valkyries need attack ups in this context. The main air is mutas, which valks destroy without upgrades, and guardians and overlords which don't shoot back. The air attack upgrade mostly affects their anti-ground, which is not taken into account in this function.

Reply:
If you don't have an ATG unit, Valks could take too much time in order to clean up the Brood Lords (once they get their armor upgrades) so your ground forces can deal with the Virophage